### PR TITLE
cockpit: Install insights-client package on demand

### PR DESCRIPTION
**Pull request by mvollmer**
- [x] #2195

### CODE REVIEW

**Comment by cnsnyder** [comment]
@candlepin-bot ok to test

File: cockpit/lib/cockpit-components-dialog.jsx [hunk 1]
```diff
@@ -45,86 +47,111 @@ let _ = cockpit.gettext;
...
             error_message: null,
         };
+        this.keyUpHandler = this.keyUpHandler.bind(this);
+        this.update_progress = this.update_progress.bind(this);
+        this.cancel_click = this.cancel_click.bind(this);
     }
+
     keyUpHandler(e) {
-        if (e.keyCode === 27) {
+        if (e.keyCode == 27) {
```
Comment (cnsnyder) [inline cockpit/lib/cockpit-components-dialog.jsx]: Why use loose equality here?

Comment (mvollmer) [reply cockpit/lib/cockpit-components-dialog.jsx]: Why not? :-)

`cockpit-component-dialog.jsx` in the Cockpit sources always has used `==` here.  The copy here got changed in 9c7c1d344628f0449ddcf1a263e2e4f80e62219d.

There are quite a number of changes like this, like changing `cancel_click(e)` to `cancel_click(event)`.  Do you want me to keep them?

**Comment by jirihnidek** [comment]
When it is not possible to download/install insight-client package (tested on CentOS 7), then following error is printed. I would print that required package is not available.

image

**Comment by jirihnidek** [comment]
I also tested it on Fedora30. The `insights-client` isn't available here too, but messaging is decent.

image

Text of error should not be aligned to right, but it should be aligned to left. I had to solve similar problem in registration dialog. So you can reuse existing CSS style for this issue.

When `insights-client` is not installed, then message is "Insights: Not connected" as you can see on following screenshot:

image

I think that there should be something like "Insights: Client not installed".

I don't think that cockpit should try to download and install insights-client automatically, when you click on: "Not connected". I would change the workflow a little bit. Cockpit plugin `cockpit-storaged` and support for NFS is IMHO nice example of good workflow for installing package (nfs-utils) in cockpit.

**Comment by jirihnidek** [comment]
@mvollmer it seems that this PR contains two commits. First commit includes refactoring of existing code and second commit contains installation of insights-client on demand. I think that we could merge first commit to master. Could it be possible to split this PR into two PRs?

**Comment by mvollmer** [comment]
> @mvollmer it seems that this PR contains two commits. First commit includes refactoring of existing code and second commit contains installation of insights-client on demand. I think that we could merge first commit to master. Could it be possible to split this PR into two PRs?

Sure: https://github.com/candlepin/subscription-manager/pull/2195

File: cockpit/lib/cockpit-components-dialog.jsx (hunk 1 above)
Comment (mvollmer) [reply cockpit/lib/cockpit-components-dialog.jsx]: This commit has been merged as #2195

**Comment by mvollmer** [comment]
> Text of error should not be aligned to right, but it should be aligned to left. I had to solve similar problem in registration dialog. So you can reuse existing CSS style for this issue.

See #2196.  I made a separate PR since this really is a bug with the dialog css.

**Comment by mvollmer** [comment]
> When it is not possible to download/install insight-client package (tested on CentOS 7), then following error is printed. I would print that required package is not available.

I think this is the honest error that happened, and I think we should show it.  Something is actually broken about the repositories or network.

**Comment by jirihnidek** [comment]
> > Text of error should not be aligned to right, but it should be aligned to left. I had to solve similar problem in registration dialog. So you can reuse existing CSS style for this issue.
> 
> See #2196. I made a separate PR since this really is a bug with the dialog css.

Thanks. It is merged into the master branch now.

**Pull request merged**